packages/www/pages/transact/eth/index.tsx

```diff
@@ -0,0 +1,454 @@
+  AlertDialogContent,
+  AlertDialogDescription,
+  AlertDialogTitle,
+  Box,
+  Button,
+  Flex,
+  Heading,
+  Label,
+  Text,
+  TextField,
+  Tooltip,
+} from "@livepeer.com/design-system";
+import { Transact as Content } from "content";
+import Layout from "layouts/main";
+
+import videoNftAbi from "./video-nft.json";
+
+const networks = {
+  "0x89": {
+    spec: {
+      chainId: "0x89",
+      chainName: "Polygon Mainnet",
+      rpcUrls: ["https://polygon-rpc.com/"],
+      nativeCurrency: { symbol: "MATIC", decimals: 18 },
+      blockExplorerUrls: ["https://polygonscan.com"],
+      iconUrls: [
+        "https://cloudflare-ipfs.com/ipfs/bafkreiduv5pzw233clfjuahv5lkq2xvjomapou7yarik2lynu3bjm2xki4",
+      ],
+    },
+    defaultContract: "0x69C53E7b8c41bF436EF5a2D81DB759Dc8bD83b5F", // TODO: Final address here
+  },
+  // "0xa4b1": {
+  //   spec: {
+  //     chainId: "0xa4b1",
+  //     chainName: "Arbitrum One",
+  //     rpcUrls: ["https://arb1.arbitrum.io/rpc"],
+  //     nativeCurrency: { symbol: "AETH", decimals: 18 },
+  //     blockExplorerUrls: ["https://arbiscan.io"],
+  //     iconUrls: [
+  //       "https://cloudflare-ipfs.com/ipfs/bafkreiamd2sujbbc673tljl7hkz66m4fqubqraq3jwnfo6smtmh6afak5i",
+  //     ],
+  //   },
+  //   defaultContract: "0xX", // TODO: Deploy a contract and add address here
+  // },
+} as const;
+const defaultNet = networks["0x89"]; // polygon
+
+async function getMintedTokenIdOnce(
+  videoNft: Contract,
+  sender: string,
+  txReceipt: TransactionReceipt
+) {
+  const events = await videoNft.getPastEvents("Mint", {
+    filter: { sender },
+    fromBlock: txReceipt.blockNumber,
+    toBlock: txReceipt.blockNumber,
+  });
+  const event = events.find(
+    (ev) => ev.transactionHash === txReceipt.transactionHash
+  );
+  return event?.returnValues?.tokenId as string;
+}
+
+async function getMintedTokenId(
+  videoNft: Contract,
+  sender: string,
+  txReceipt: TransactionReceipt,
+  logger: (log: JSX.Element | string) => void
+) {
+  const maxAttempts = 5;
+  const retryDelayMs = 5 * 1000;
+  const sleep = (ms: number) => new Promise((res) => setTimeout(res, ms));
+
+  let lastErr: Error | undefined;
+  for (let a = 1; a <= maxAttempts; a++) {
+    if (a > 1) {
+      await sleep(retryDelayMs);
+    }
+    try {
+      const tokenId = await getMintedTokenIdOnce(videoNft, sender, txReceipt);
+      if (!tokenId) {
+        throw new Error("Missing tokenId");
+      }
+      return tokenId;
+    } catch (err) {
+      lastErr = err;
+    }
+  }
+  logger(`Error getting minted token ID: ${lastErr}`);
+  return null;
+}
+
+async function mintNft(
+  web3: Web3,
+  contractAddress: string,
+  from: string,
+  to: string,
+  tokenUri: string,
+  logger: (log: JSX.Element | string) => void
+) {
+  try {
+    logger("Started mint transaction...");
+    const videoNft = new web3.eth.Contract(videoNftAbi as any, contractAddress);
+    const transaction = {
+      to: contractAddress,
+      gas: 500000,
```

Comment:
Maybe it should calculate the estimated gas limit for this contract payable call (I think with web3.eth.estimateGas?)

Reply:
ideally it should, but the estimated gas is apparently really bad and the transactions take so long to execute on Polygon. Although maybe that was only a problem cause I was testing on mainnet and wanted to iterate quickly 🙈 

I'm gonna see what can be done here, thanks for pointing out the function for estimating it (I tried it by not suggesting anything and letting metamask do it)

Reply:
Mmm strange. I wonder if it's a behavior related to Polygon. In any case, I would think that letting the user input the gas limit would be better anyway than setting a very high default. Or maybe we could manually calculate it on our function call and hardcode the correct one.

Could possibly be that estimateGas is failing because we are also specifying the gasPrice, I would try to remove also that and retry with the estimateGas.

Another thing, maybe we could try to calculate the gas directly from the contract, like this?

```
const contract = web3.eth.contract(abi).at(address);
const gasEstimate = contract.$myFunctionMethod.estimateGas(arg1, arg2, {from: account});
```

Reply:
I think Yondon could help on this matter

Reply:
> I would think that letting the user input the gas limit would be better anyway than setting a very high default. Or maybe we could manually calculate it on our function call and hardcode the correct one.

Oh, sorry. I thought you were talking about the gas price in the whole conversation. This gas limit has been chosen from the transaction that we run here. It's set to ~2x the gas that is usually spent just in case, since our contract implementation could also change in the future (especially if we do a proper openzeppeling proxy thing so we are able to update it).


The gas price thing that was confusing and the values suggested by MetaMask were bad.

But nevertheless, none of those values are final. The user will still choose on the MetaMask popup how much they want to pay for gas and what is the max gas allowed to be spent.

Reply:
Oh cool, so the gasLimit is already set based on the contract function! :) nice, then that's great. 

About the gasPrice, yeah, is not always reliable. There could be an "advanced tab" but I don't think it's something we should cover right now, and MetaMask let you customize the gas anyway, if needed, so yeah I agree